prereise/gather/griddata/hifld/data_process/demand.py

```diff
@@ -0,0 +1,81 @@
+import inspect
+import os
+
+import pandas as pd
+
+from prereise.gather.griddata import hifld
+from prereise.gather.griddata.hifld import const
+
+
+def assign_demand_to_buses(substations, branch, plant, bus):
+    """Using data on population by county and ZIP code, assign demand to substations,
+    then to the lowest-voltage bus within each substation.
+    This information is added inplace as a 'Pd' column to the ``bus`` data frame.
+
+    :param pandas.DataFrame substations: table of substation data.
+    :param pandas.DataFrame branch: table of branch data.
+    :param pandas.DataFrame plant: table of plant data.
+    :param pandas.DataFrame bus: table of bus data.
+    """
+    # Load data
+    data_dir = os.path.join(os.path.dirname(inspect.getfile(hifld)), "data")
```

Comment:
Can we simply do `data_dir = os.path.join(os.path.dirname(os.path.dirname(__file__), "data")` here?

Reply:
For now, this would be equivalent, but I figured that the location of the data directory would be more stable if it's defined as relative to the `prereise.gather.griddata.hifld` directory, rather than relative to the location of this chunk of code (which might be moved in a refactor).

Reply:
Agree. However, we will need to import the `hifld` module just for the `inspect` call here. Maybe even we move this chunk of code around, it might still end up with two hops from the root? Anyway, will defer to your decision here.

Reply:
I bet you're right that this code will probably stay in a compatible place, but I prefer the more stable version, even if it's a little more verbose.